src/java.base/share/classes/sun/security/util/KeyUtil.java

```diff
@@ -97,6 +98,16 @@ public static final int getKeySize(Key key) {
         } else if (key instanceof DHKey) {
             DHKey pubk = (DHKey)key;
             size = pubk.getParams().getP().bitLength();
+        } else if (key instanceof XECKey) {
+            String name = ((NamedParameterSpec)((XECKey) key).getParams()).getName();
```

Comment:
I hope the params is always `NamedParameterSpec` but would rather be safe to check instanceof first.

Reply:
Added the checking.